### PR TITLE
netdev: make netdev_% variants a submodule

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR makes `netdev` a module and add `netdev_%` variants as submodules.
This should help to add new netdev components (`netdev_lora`?) and helpers associated to `netdev` (and not a variant)

### Testing procedure
Try compiling `gnrc_networking` for `native` and any IEEE802.15.4 board like `samr21-xpro`

### Issues/PRs references
#13562

### CODE REVIEW

**Comment by jia200x** [comment]
There was a whitespace. I fixed it and amended directly.

**Comment by miri64** [comment]
*something* broke ;-)

**Comment by jia200x** [comment]
I moved `netdev_tap` to the `netdev` folder. It should work now.

File: drivers/Makefile.dep [hunk 1]
```diff
@@ -737,3 +737,7 @@ ifneq (,$(filter xbee,$(USEMODULE)))
   USEMODULE += xtimer
   USEMODULE += netif
 endif
+
+ifneq (,$(filter netdev_%,$(USEMODULE)))
```
Comment (miri64) [inline drivers/Makefile.dep]: This line and ...

File: makefiles/pseudomodules.inc.mk [hunk 2]
```diff
@@ -57,7 +57,7 @@ PSEUDOMODULES += log_color
 PSEUDOMODULES += lora
 PSEUDOMODULES += mpu_stack_guard
 PSEUDOMODULES += nanocoap_%
-PSEUDOMODULES += netdev_default
+PSEUDOMODULES += netdev_%
```
Comment (miri64) [inline makefiles/pseudomodules.inc.mk]: ... this line are the _actual_ problem. They are just declaring everything starting with `netdev_` a submodule of `netdev`. However, `netdev_tap` is not. It's just a `netdev` simulator using the hosts TAP interface (where the name is basically coming from). I would prefer, to either name the submodules explicitly, or do it in a similar way that I already tried in https://github.com/RIOT-OS/RIOT/pull/10970.

**Comment by jia200x** [comment]
I just added it to the "NO_PSEUDOMODULES" list and amended directly. I removed the `netdev_tap` commit.

File: drivers/Makefile.dep (hunk 1 above)
Comment (jia200x) [reply drivers/Makefile.dep]: I would keep this line anyway. In worst case it doesn't compile any module, but I plan to add some common helpers for `netdev` that might be used by `netdev_tap` too.

**Comment by aabadie** [comment]
It seems that there's consensus on this one. Just need a rebase!

**Comment by aabadie** [comment]
ping @jia200x, please rebase!

**Comment by aabadie** [comment]
ping @jia200x, please rebase.

Just thinking of the `netdev_tap` module name, would it makes sense to call it `native_tap` instead ? This way it would be pretty clear that it's a "tap" driver and that it's only for native.

**Comment by jia200x** [comment]
> Just thinking of the netdev_tap module name, would it makes sense to call it native_tap instead ? This way it would be pretty clear that it's a "tap" driver and that it's only for native.

Yes! Definitely. The `netdev_tap` is simply an implementation of a native driver that runs TAP, on top of `netif`. I would be in favor of this.

**Comment by jia200x** [comment]
I changed the logic a little bit. I'm not touching `netdev_tap` now (although I agree with @aabadie that we should change the name).

**Comment by jia200x** [comment]
I piggybacked the fix here

File: Makefile.dep [hunk 3]
```diff
@@ -191,6 +186,11 @@ ifneq (,$(filter gnrc_netif,$(USEMODULE)))
   USEMODULE += netif
   USEMODULE += l2util
   USEMODULE += fmt
+
+  # Don't register netdevs if there is only a single one of them
+  ifeq (,$(filter gnrc_netif_single,$(USEMODULE)))
+    USEMODULE += netdev_register
+  endif
```
Comment (benpicco) [inline Makefile.dep:193]: netdev registration is independent of `gnrc` though.
We should still get the same MAC addresses with LWIP or OpenThread.

Comment (jia200x) [reply Makefile.dep:193]: Ok, I understand now the rationale behind it. But I think this module should be pulled when there's a user. It should at least check if there's someone trying to pull `netdev`. Now this should be possible with this PR

Comment (jia200x) [reply Makefile.dep]: Done! (see https://github.com/RIOT-OS/RIOT/pull/13565/commits/ad61f2852da2eb15bc0fcddab1ffe7203e3f6c92)

Comment (benpicco) [reply Makefile.dep]: Looks good to me!

**Comment by jia200x** [comment]
I refreshed the labels. It should be green soon :)

**Comment by jia200x** [comment]
thanks for the review!

**Pull request merged**